### PR TITLE
Port over migration and tool use of centrifuges

### PR DESCRIPTION
**Pull request by chaosvolt**
#### Summary

SUMMARY: Infrastructure "Port over updates to centrifuge terrain, port over ability to use centrifuge as a tool"

#### Purpose of change

Follow-up to https://github.com/cataclysmbnteam/Cataclysm-BN/pull/1374, since what was needed to get centrifuges as tools basically required a bit more work than initially intended.

#### Describe the solution

1. Partially port over DDA conversion of utility terrain into furniture, specifically only centrifuges. This is because turns out terrain can't handle `crafting_pseudo_item`, and this would be easier.
2. Obsoleted terrain version of centrifuge. json/obsoletion/terrains.json is evidently for overmap terrain, not terrain-type terrrain, so put it in uncategorized.json instead.
3. Updated looks-like of microcentifuge.
4. Replaced centifuge terrain with furntiure in lab basement, lab floorplans, 
5. Updated FEMA camp, hosptial, and lab mapgen palettes accordingly.
6. Updated centrifuge spawns in mods. Lel, I actually can't find any sign that anyone ever bothered to do that in DDA. Good example can be seen in Dinomod: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/DinoMod/mapgen/DinoLabFinale.json#L41 and a modern Aftershock example: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/Aftershock/maps/mapgen/millyficent_lab.json#L245
7. Ported over fix to computer_session.cpp that correctly looks for centrifuge furniture in the blood analysis mission.
8. Re-added change to centrifuge furniture description from the previous PR's first commit.
9. Ported over use of `crafting_pseudo_item` for centrifuge.
10. Ported over tool entry for fake centrifuge, with same qualities as microcentrifuge now has.

#### Describe alternatives you've considered

Hacking together support for terrain being able to use `crafting_pseudo_item`, was reluctant to explore it just yet.

#### Testing

1. Checked affected files for syntax and lint errors.
2. Compiled and load tested.
3. Zooped over to a hospital to confirm that centrifuge still works.

#### Additional context

Sources:
1. Migration of centrifuge from terrain to furniture, by I-am-Erk: https://github.com/CleverRaven/Cataclysm-DDA/pull/41731
2. Fixing what the above PR broke, by johnstoecker: https://github.com/CleverRaven/Cataclysm-DDA/pull/43361
3. Making centrifuges useful for separation, by zachary-kaelan: https://github.com/CleverRaven/Cataclysm-DDA/pull/34344

Mapdata.cpp and mapdata.h still have references to `t_centrifuge` in DDA, should we remove them in BN or are they still important for something?

### CODE REVIEW

File: data/json/obsoletion/uncategorized.json [hunk 1]
```diff
@@ -363,5 +363,31 @@
       "install": { "skills": [ [ "mechanics", 5 ], [ "electronics", 5 ] ] },
       "removal": { "skills": [ [ "mechanics", 3 ] ] }
     }
+  },
+  {
+    "type": "terrain",
+    "id": "t_centrifuge",
+    "name": "centrifuge",
+    "description": "This is a centrifuge, a liquid separating device with an automated analyzer unit.  It could be used to analyze a medical fluid sample, such as blood, if a test tube was placed in it.",
```
Comment (Coolthulhu) [inline data/json/obsoletion/uncategorized.json:371]: Would be nice if the old one had a note that explains why they can't be used. "Broken" in name or something like that.

File: src/computer_session.cpp [hunk 2]
```diff
@@ -745,7 +745,7 @@ void computer_session::action_blood_anal()
 {
     g->u.moves -= 70;
     for( const tripoint &dest : g->m.points_in_radius( g->u.pos(), 2 ) ) {
-        if( g->m.ter( dest ) == t_centrifuge ) {
+        if( g->m.furn( dest ) == furn_str_id( "f_centrifuge" ) ) {
```
Comment (Coolthulhu) [inline src/computer_session.cpp:748]: Would be better with this declared near the top of the file.

Comment (chaosvolt) [reply src/computer_session.cpp:748]: Looking for a good example of furniture being declared like that in a file, hard to find a case where it's set up like that.

Comment (Coolthulhu) [reply src/computer_session.cpp:748]: ter/furn ids are weird like that. I think we'll need a header for this crap, it's messy.
For now, it's no big deal.

**Pull request merged**